pkg/state/state.go

```diff
@@ -20,6 +21,7 @@ type State struct {
 	Azure        azure.Azure
 	CloudStack   cloudstack.CloudStack
 	DigitalOcean digitalocean.DigitalOcean
+	External     external.External
```

Comment:
just a nit: but maybe the provider name could be more descriptive to indicate that it's only terraform related? A suggestion: `TerraformExternal`

It does feel a little odd to include this here TBH, but then again the only "providers" we have so far are "service related" like AWS, Azure, etc.

What do you think?

Reply:
Yeah I had a similar uncertainty. However it is actually a provider, not something built-in to terraform stdlib: https://registry.terraform.io/providers/hashicorp/external/latest/docs 
So if anything different it should be `HashicorpExternal` based on the provider namespace?

I don't have a strong opinion on what it should be, though I will note I am working on another PR that does add internal Terraform structures (imported modules, provisioners, etc) so they can be queried/matched agains with rego if that changes your opinon on how this one should be named. I was thinking that one will be just `Terraform` here with child resources like `Modules` and `Provisioners` within.

Reply:
> I don't have a strong opinion on what it should be, though I will note I am working on another PR that does add internal Terraform structures (imported modules, provisioners, etc) so they can be queried/matched agains with rego if that changes your opinon on how this one should be named. I was thinking that one will be just `Terraform` here with child resources like `Modules` and `Provisioners` within.

Yeah that sounds good to me - what about a `Hashicorp` package that includes this provider and the other `Terraform` related things (child resources like modules and provisioners as you mentioned).